src/rsa/bigint.rs

```diff
@@ -1052,6 +1058,7 @@ mod repr_c {
 
     // Keep in sync with `bignum_st` in openss/bn.h.
     #[repr(C)]
+    #[derive(Clone)]
```

Comment:
We cannot derive `Clone` for this type because `d` is a pointer. In order to clone `d` we need to make a copy of the data being pointed to by `d`.

Reply:
I suggest that you split this PR into two parts: One for RSA and one for non-RSA stuff. The RSA stuff is tricky and the non-RSA stuff is simple.

The `Nonnegative` type already implements a variant of `Clone::clone()` called `try_clone()`. We were expecting that there would be a `TryClone` trait added to the standard library, but it looks like that won't be the case; see https://github.com/rust-lang/rfcs/pull/2189#issuecomment-381439136. At least at the present time I don't want to change *ring* to panic when the C functions return errors, so I think we cannot implement `Clone` for RSA private keys. If you want an RSA private key that you can `Clone::clone()` in your app then you can get one by using `Arc<RSAKeyPair>` instead of `RSAKeyPair`.